src/test/fuzz/coinscache_sim.cpp

```diff
@@ -148,8 +146,10 @@ class CoinsViewBottom final : public CCoinsView
 public:
     std::optional<Coin> GetCoin(const COutPoint& outpoint) const final
     {
-        // TODO GetCoin shouldn't return spent coins
-        if (auto it = m_data.find(outpoint); it != m_data.end()) return it->second;
+        if (auto it = m_data.find(outpoint); it != m_data.end()) {
+            assert(!it->second.IsSpent());
+            return it->second;
+        }
         return std::nullopt;
     }
 
```

Comment:
We should be able to remove `HaveCoin` now, since
```C++
bool CCoinsView::HaveCoin(const COutPoint &outpoint) const
{
    return GetCoin(outpoint).has_value();
}
```
is more accurate (especially with the new extra assert)

Reply:
Done.

Reply:
Isn't that the same as the parent in `CCoinsView`?

Reply:
Ah right, removed.